src/StateMachine.cpp

```diff
@@ -22,8 +22,9 @@ StateMachine::Dictionary::iterator StateMachine::declareInternal(uint16_t object
     ).first;
 }
 
-StateMachine::StateMachine(uint8_t nodeId)
+StateMachine::StateMachine(uint8_t nodeId, bool useUnknownSizes)
     : nodeId(nodeId)
+    , useUnknownSizes(useUnknownSizes)
```

Comment:
Does it really make sense to have the flag globally ? I would assume that it is per-object.

Reply:
I don't think devices do that per object. I'm certain the one I'm currently working on never sets the size bits so only globally makes sense.

Reply:
The library concept is to define an object library for a particular device. With the per-object extension I've outlined below, it should be feasible.

What's unclear to me is whether the objects are variable size (i.e. you really have to read the SDO to read the object), or if it is a traditional SDO object dictionary, just without sizes in the SDO replies ? 

In the first case, I think the best would be to delegate the size decoding to a subclass (through a virtual method)

In the second case, as an alternative to [this proposal](https://github.com/rock-drivers/drivers-canopen_master/pull/5#discussion_r425112891), I believe that a good enough solution would be to force-set the size of the command from the known size of the object before calling `setObjectValue`.

Reply:
Exactly what I explained in the other thread. Your thinking was quicker than my explaining :P

Reply:
> it is a traditional SDO object dictionary, just without sizes in the SDO replies ?

Exactly.

> I believe that a good enough solution would be to force-set the size of the command from the known size of the object before calling setObjectValue

I agree